cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -14,7 +14,7 @@ namespace gluten {
 
 namespace {
 const std::string kSparkBatchSizeKey = "spark.sql.execution.arrow.maxRecordsPerBatch";
-const std::string kSparkOffHeapSizeKey = "spark.memory.offHeap.size";
+const std::string kSparkOffHeapMemoryPerTask = "spark.gluten.memory.offHeap.size.in.bytes.pertask";
```

Comment:
@rui-mo 

I see you have set 0.75 * executor-offheap-size as `kMaxPartialAggregationMemory`.

https://github.com/oap-project/gluten/blob/15faa98cda0a1a2fe8f7c2b3e4c73ce159b791f7/cpp/velox/compute/WholeStageResultIterator.cc#L182-L190

In this PR I am changing `kSparkOffHeapSizeKey` to `kSparkOffHeapMemoryPerTask` (per executor -> per task). Should we update the factor 0.75 to another number? Was it intended to use executor's off-heap memory size for partial agg?

Reply:
This setting was used to enlarge the default value and to achieve more flexible flush, see https://github.com/oap-project/velox/pull/98. Looks an adjustment is needed with this PRs' change but I'm not sure about the value. Do you have any suggestion? @FelixYBW

Reply:
I restored the old code of this part and left a comment there. Let's propose fix later on.